ocsp_status.go

```diff
@@ -108,74 +146,117 @@ func getOCSPResponse(rootCert *x509.Certificate, issuerCert *x509.Certificate, r
 		}
 	}
 
-	parsedResp, err := ocsp.ParseResponse(ocspResp, issuerCert)
+	return ocspResp
+}
+
+func lintFile(respFile string) {
+	ocspResp, err := ioutil.ReadFile(respFile)
+	if err != nil {
+		panic("Error reading OCSP Response file: " + err.Error())
+	}
+	// can't check signature w/o outside knowledge of who the issuer should be
+	// TODO: add functionality so user can specify who the issuer is
+	parsedResp, err := ocsp.ParseResponse(ocspResp, nil)
 	if err != nil {
 		fmt.Println(string(ocspResp))
 		panic(err.Error())
 	}
+	linter.CheckOCSPResp(parsedResp)
+}
 
-	return parsedResp
+func parseAndLint(ocspResp []byte, issuerCert *x509.Certificate) error {
+	parsedResp, err := ocsp.ParseResponse(ocspResp, issuerCert)
+	if err != nil {
+		fmt.Println(string(ocspResp)) // for debugging, will remove
+		return err
+	}
+	linter.CheckOCSPResp(parsedResp)
+	return nil
+}
+
+func checkFromCert(certFile string, get bool, ocspURL string, dir string, hash crypto.Hash) error {
+	reqMethod := http.MethodPost
+	if get {
+		reqMethod = http.MethodGet
+	}
+
+	ocspReq, issuerCert := createOCSPReqFromCert(certFile, ocspURL, reqMethod, hash)
+	ocspResp := getOCSPResponse(ocspReq, dir)
+
+	return parseAndLint(ocspResp, issuerCert)
+
+}
+
+func checkFromURL(serverURL string, print bool, get bool, ocspURL string, dir string, hash crypto.Hash) error {
+	tlsConn := createConn(serverURL)
+
+	certChain := tlsConn.ConnectionState().PeerCertificates
+	rootCert := certChain[0]
+	issuerCert := certChain[len(certChain)-1]
+
+	if print {
+		printCert(rootCert)
+	}
+
+	ocspResp := tlsConn.OCSPResponse()
+	tlsConn.Close()
```

Comment:
does the connection close always happen? Does it need defer?

Reply:
I should probably ask Ryan or Filippo about this